thumbnail-service/src/main/java/no/sikt/nva/handler/ThumbnailRequestHandler.java

```diff
@@ -0,0 +1,144 @@
+        this.s3Client = s3Client;
+    }
+
+    @Override
+    public Void handleRequest(S3Event s3Event, Context context) {
+        inputFile = readFile(s3Event);
+        var outputFile = generateThumbnail();
+        writeThumbnailToS3(outputFile);
+        outputFile.deleteOnExit();
+        inputFile.deleteOnExit();
+        return null;
+    }
+
+    private void writeThumbnailToS3(File outputFile) {
+        try {
+            s3Client.putObject(PutObjectRequest.builder()
+                                   .bucket(thumbnailBucketName)
+                                   .key("keyForFeature(featureState.getFeature())")
+                                   .build(), RequestBody.fromFile(outputFile));
+        } catch (RuntimeException e) {
+            logger.warn(e.getMessage());
+            throw new RuntimeException(e);
+        }
+    }
+
+    private File generateThumbnail() {
+        try (var thumbnailerManager = new ThumbnailerManager()) {
+            logCurrentSupportedMimeTypes(thumbnailerManager);
+            var outPutFile = new File(INPUT_FILE_NAME_PREFIX + OUTPUT_FILE_NAME);
+            thumbnailerManager.generateThumbnail(inputFile, outPutFile, mimeTypeFromS3Response);
+            return outPutFile;
+        } catch (IOException | ThumbnailerException e) {
+            logger.warn(COULD_NOT_CREATE_THUMBNAIL_LOG_MESSAGE);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void logCurrentSupportedMimeTypes(ThumbnailerManager thumbnailerManager) {
+        var supportedMimeTypes = String.join(DELIMITER, thumbnailerManager.getAcceptedMimeTypes());
+        logger.info(CURRENTLY_SUPPORTING_THE_FOLLOWING_MIME_TYPES_LOG_MESSAGE + supportedMimeTypes);
+    }
+
+    private File readFile(S3Event s3Event) {
+        var objectKey = getObjectKey(s3Event);
+        var bucketName = getBucketName(s3Event);
+        var objectRequest = createObjectRequest(objectKey, bucketName);
+        return attempt(() -> s3Client.getObject(objectRequest))
+                   .map(this::readResponseAndRetrieveMimeType)
+                   .orElseThrow(this::logFailureAndThrowException);
+    }
+
+    private RuntimeException logFailureAndThrowException(Failure<File> failure) {
+        logger.warn(failure.getException().getMessage());
+        throw new RuntimeException(failure.getException());
+    }
+
+    private File readResponseAndRetrieveMimeType(ResponseInputStream<GetObjectResponse> response) {
+        mimeTypeFromS3Response = response.response().contentType();
+        String inputFileName = determineFileName(response);
+        File inputFile = new File(inputFileName);
+        try (var fileOutputStream = Files.newOutputStream(inputFile.toPath())) {
+            fileOutputStream.write(response.readAllBytes());
+        } catch (Exception e) {
+            logger.warn(e.getMessage());
+            throw new RuntimeException(e);
```

Comment:
Usually exceptions are logged already. Are we sure that we do not log here twice?

Reply:
fixed